cmd/skopeo/sync.go

```diff
@@ -74,6 +75,25 @@ type registrySyncConfig struct {
 // sourceConfig contains all registries information read from the source YAML file
 type sourceConfig map[string]registrySyncConfig
 
+// syncConfig contains all registries information read from the source YAML file for source yaml2
+type syncConfig map[string]registrySyncConfigV2
```

Comment:
- So is this `syncConfigV2`?

- One of the major lessons from v1 is that having a top-level `map[something]something` structure makes it impossible to add options at that level. So maybe the top-level should be a `struct` here as well.